waku/v2/node/wakunode2_setup_metrics.nim

```diff
@@ -24,24 +24,44 @@ proc startMetricsServer*(serverIp: ValidIpAddress, serverPort: Port) =
 
     info "Metrics HTTP server started", serverIp, serverPort
 
+proc parseCollectorIntoF64(collector: Collector): float64 = 
+  var total = 0.float64
+  for key in collector.metrics.keys():
+    try:
+      total = total + collector.value(key)
+    except KeyError:
+      discard
+  return total
 
 proc startMetricsLog*() =
   # https://github.com/nim-lang/Nim/issues/17369
   var logMetrics: proc(udata: pointer) {.gcsafe, raises: [Defect].}
 
+  var cumulativeErrors = 0.float64
+  var cumulativeConns = 0.float64
+
   logMetrics = proc(udata: pointer) =
     {.gcsafe.}:
       # TODO: libp2p_pubsub_peers is not public, so we need to make this either
       # public in libp2p or do our own peer counting after all.
-      var totalMessages = 0.float64
-      for key in waku_node_messages.metrics.keys():
-        try:
-          totalMessages = totalMessages + waku_node_messages.value(key)
-        except KeyError:
-          discard
-
-    info "Node metrics", totalMessages
-    discard setTimer(Moment.fromNow(2.seconds), logMetrics)
+
+      var totalErrors = parseCollectorIntoF64(waku_node_errors)
+      var totalConnections = parseCollectorIntoF64(waku_node_conns_initiated)
+
+      # track cumulative, and then max.
+      var freshErrorCount = max(totalErrors - cumulativeErrors, 0)
+      var freshConnCount = max(totalConnections - cumulativeConns, 0)
```

Comment:
If you are not modifying the variable value within the scope, please use let. It helps to know what is modified and what is not at first sight. And if you change mistakenly a `let` the compiler will warn you.

Reply:
Will do, thanks!

Reply:
resolved in https://github.com/status-im/nwaku/pull/1094/commits/848ca0937f6de6ea7b387f1f0870c6273414cb9a